### DISCUSSION TITLE
FeatureSet - validation for entities and features

### DISCUSSION THREAD

**Discussion by george0st** [mlrun/mlrun#1858] [Q&A] [answered]
Hello @Hedingber, @katyakats , I have question about validation of names, types for features and entities. Do you see as good idea to add these validations to the methods add_features, add_entities? 

In case of name validation, is it fine to check only these restrictions or others? I think:
 - 't_*' or '_*'
 - 'aggr_*'
 - name length?

BTW: 

- In current situation (without validation), it can generate issues in case of aggregation, etc. 
- I attached a few "valid" samples which I tested in MLRun 0.7.1
![Featureset-Save01](https://user-images.githubusercontent.com/95856749/161212160-4d11dfd2-9989-4125-b749-20c9e0d340e6.png)
![Featureset-Save02](https://user-images.githubusercontent.com/95856749/161212183-83e005fe-4f9d-4d96-837b-35e850cc42b1.png)

**Comment by katyakats** [comment, marked as answer]
sounds good to me (after the release). @theSaarco what do you think?

**Reply by theSaarco** [reply]
:+1: I like this proposal - from my knowledge of the code I'm not concerned about SQL injection, but "strange names" might pose issues when performing Spark queries, for example (column names may be invalid etc.). We should also add tests to check these things.
We should at a minimum enforce limitations on the names of the features/entities and the types of entities - `Entity` actually accepts `value_type: ValueType` as input but there's no validation at all on that.

**Comment by george0st** [comment]
@katyakats , I saw that for aggregation you change prefix from 't_' to '_'. Do you plan others changes (in near future) e.g. do you plan to split user columns and added system columns (based on internal processing or aggregation)?

@theSaarco , I would like to show, what can happend (e.g. SQL injection, etc.) but important is to check types also, because it can generate unexpected behavioral. I am not sure if it can affect also MySQL DB (as main repository for MLRun), but probably penetration tests can provide the answer.

**Reply by katyakats** [reply]
@george0st no
